core/trino-main/src/main/java/io/trino/execution/QueryTracker.java

```diff
@@ -259,12 +259,11 @@ private void failAbandonedQueries()
 
                 if (isAbandoned(query)) {
                     log.info("Failing abandoned query %s", query.getQueryId());
-                    query.fail(new TrinoException(
-                            ABANDONED_QUERY,
-                            format("Query %s has not been accessed since %s: currentTime %s",
-                                    query.getQueryId(),
-                                    query.getLastHeartbeat(),
-                                    DateTime.now())));
+                    query.fail(new TrinoException(ABANDONED_QUERY, format(
+                            "Query %s was abandoned by the client, as it may exited or stopped checking for query results. Query results have not been accessed since %s: currentTime %s",
```

Comment:
minor english grammar thing ... I don't think `may` goes with past tense without `have`. we can change `may exited or stopped checking` to `either exited or stopped checking`, if you don't want to use `may *have* exited`. cc native english speaker @electrum

Reply:
right. changing `may exited` to `may have exited`.